icechunk/src/ops/stats.rs

```diff
@@ -0,0 +1,75 @@
+use futures::TryStreamExt as _;
+use std::{collections::HashSet, sync::Arc};
+use tokio::pin;
+
+use crate::{
+    Storage,
+    asset_manager::AssetManager,
+    format::{IcechunkFormatError, IcechunkFormatErrorKind, manifest::ChunkPayload},
+    repository::RepositoryResult,
+    storage,
+};
+
+use super::pointed_snapshots;
+
+/// Compute the total size in bytes of all committed repo chunks.
+/// It doesn't include inline or virtual chunks.
+pub async fn repo_chunks_storage(
+    storage: &(dyn Storage + Send + Sync),
+    storage_settings: &storage::Settings,
+    asset_manager: Arc<AssetManager>,
+) -> RepositoryResult<u64> {
+    let extra_roots = Default::default();
+    let all_snaps = pointed_snapshots(
+        storage,
+        storage_settings,
+        Arc::clone(&asset_manager),
+        &extra_roots,
+    )
+    .await?;
+
+    let mut seen_manifests = HashSet::new();
+    let mut seen_chunks = HashSet::new();
+    let mut size = 0;
+
+    pin!(all_snaps);
+    while let Some(snap_id) = all_snaps.try_next().await? {
+        let snap = asset_manager.fetch_snapshot(&snap_id).await?;
+
+        for manifest_id in snap.manifest_files().map(|mf| mf.id) {
```

Comment:
Is it worth extracting this to be able to get the size of a specific snapshot? Or is that not quite relevant. Wondering if it could be helpful for deciding when to gc

Reply:
Yeah, that would be useful, but I don't have an immediate use case. I can add it a bit down the road if that's OK